src/rebar_prv_eunit.erl

```diff
@@ -167,15 +207,44 @@ replace_src_dirs(State) ->
     StrippedOpts = lists:keydelete(src_dirs, 1, ErlOpts),
     rebar_state:set(State, erl_opts, [{src_dirs, ["test"]}|StrippedOpts]).
 
-maybe_cover_compile(State, Opts) ->
-    State1 = case proplists:get_value(cover, Opts, false) of
-        true  -> rebar_state:set(State, cover_enabled, true);
-        false -> State
-    end,
-    rebar_prv_cover:maybe_cover_compile(State1).
+test_set(Apps, Suites) -> test_set(Apps, Suites, []).
+
+test_set([], all, Acc) -> lists:reverse(Acc);
+test_set(_, [], Acc) -> lists:reverse(Acc);
+test_set([App|Rest], all, Acc) ->
+    AppName = list_to_atom(binary_to_list(rebar_app_info:name(App))),
+    test_set(Rest, all, [{application, AppName}|Acc]);
+test_set(Apps, [Suite|Rest], Acc) ->
+    test_set(Apps, Rest, [{module, list_to_atom(Suite)}|Acc]).
```

Comment:
Wouldn't this function be clearer if you just split it up in like `test_set_all` and `test_set_suites` or something? It seems both are entirely disjoint, but fuzed in the same function.